ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/PartyMapper.java

```diff
@@ -70,34 +72,56 @@ public CCDParty to(Party party) {
     }
 
     @Override
-    public Party from(CCDParty ccdParty) {
+    public Party from(Object party) {
+        CCDParty ccdParty = (CCDParty) party;
         switch (ccdParty.getType()) {
             case COMPANY:
                 CCDCompany ccdCompany = ccdParty.getCompany();
-                return new Company(ccdCompany.getName(), addressMapper.from(ccdCompany.getAddress()),
+                return new Company(ccdCompany.getName(),
+                    addressMapper.from(ccdCompany.getAddress()),
                     addressMapper.from(ccdCompany.getCorrespondenceAddress()),
-                    ccdCompany.getPhoneNumber(), representativeMapper.from(ccdCompany.getRepresentative()),
+                    ccdCompany.getPhoneNumber(),
+                    representativeMapper.from(ccdCompany.getRepresentative()),
                     ccdCompany.getContactPerson());
             case INDIVIDUAL:
                 CCDIndividual ccdIndividual = ccdParty.getIndividual();
-                return new Individual(ccdIndividual.getName(), addressMapper.from(ccdIndividual.getAddress()),
-                    addressMapper.from(ccdIndividual.getCorrespondenceAddress()), ccdIndividual.getPhoneNumber(),
+                return new Individual(ccdIndividual.getName(),
+                    addressMapper.from(ccdIndividual.getAddress()),
+                    addressMapper.from(ccdIndividual.getCorrespondenceAddress()),
+                    ccdIndividual.getPhoneNumber(),
                     representativeMapper.from(ccdIndividual.getRepresentative()),
-                    LocalDate.parse(ccdIndividual.getDateOfBirth(), DateTimeFormatter.ISO_DATE));
+                    parseDob(ccdIndividual.getDateOfBirth())
+                );
             case SOLE_TRADER:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 9 to at most 5, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1151)